pkg/controller/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -205,3 +226,65 @@ func intstrPtr(i int32) *intstr.IntOrString {
 	intstr := intstr.FromInt(int(i))
 	return &intstr
 }
+
+func expectReconcile(t *testing.T, requests chan reconcile.Request, errors chan error) {
+	t.Helper()
+
+	{
+		ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+		defer cancel()
+
+	LOOP0:
+		for range time.Tick(pollingInterval) {
+			select {
+			case recv := <-requests:
+				if recv == expectedRequest {
+					break LOOP0
+				}
+			case <-ctx.Done():
+				t.Errorf("timed out waiting reconcile")
+				break LOOP0
+			}
+		}
+	}
+
+	{
+		ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+		defer cancel()
+
+	LOOP1:
+		for range time.Tick(pollingInterval) {
+			select {
+			case err := <-errors:
+				if err == nil {
+					break LOOP1
```

Comment:
same question about this break; why start over instead of exiting?

Reply:
The second loop (`LOOP1`) could just return, but I kept the semantics the same as the first loop for consistency. This will change when I do each loop concurrently.